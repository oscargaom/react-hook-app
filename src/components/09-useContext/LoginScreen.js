import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const handleLogin = () => {
        setUser({
            id:9801,
            name:"oscar",
            email:"oscar@gmail.com"
        });
    };

    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>

            <button
                className="btn btn-primary"
                onClick={handleLogin}
            >
                Login
            </button>
        </div>
    )
}
