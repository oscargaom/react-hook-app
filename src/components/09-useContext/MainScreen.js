import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export const MainScreen = () => {

    const [user, setUser] = useState({
        id: 0,
        name: "",
        email:""
    });

    return (
        <UserContext.Provider value={{
            user,
            setUser
        }}>
            <AppRouter />
        </UserContext.Provider>
    )
}
