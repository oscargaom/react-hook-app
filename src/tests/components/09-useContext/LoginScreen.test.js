import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';


describe('Pruebas del componente <LoginScreen />', () => {
    
    const user = {};
    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={
            {
                user,
                setUser
            }
        }>
            <LoginScreen />
        </UserContext.Provider>
    );

    test('debe mostrarse correctamente ', () => {
        expect(wrapper).toMatchSnapshot();    
    });

    test('debe ejecutar el setUser con el argumento indicado ', () => {
        wrapper.find('button').simulate('click');

        expect(setUser).toHaveBeenCalledTimes(1);
        expect(setUser).toHaveBeenCalledWith(expect.any(Object));
        expect(setUser).toHaveBeenCalledWith({
            id:9801,
            name:"oscar",
            email:"oscar@gmail.com"
        });

        // console.log(wrapper);
    });
    
});
