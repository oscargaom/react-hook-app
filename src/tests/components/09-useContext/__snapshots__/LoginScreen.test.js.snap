// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <LoginScreen /> debe mostrarse correctamente  1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
